### ISSUE TITLE
xmr-stak-cpu not using all of my cores

### ISSUE THREAD

**Issue by Yoshi24517** [fireice-uk/xmr-stak#1813]
I am using Google Cloud Platform to run xmr-stak. (I have permission to do this, as I have an upgraded account; upgraded accounts are allowed to mine.)

I have 24 vCPUs/cores, all Intel-Haswell processors. No AMD or NVIDIA GPUs.
Running Ubuntu 16.04 LTS.

Everything has compiled successfully, I'm able to mine, but, when I check my hashrate:

```
HASHRATE REPORT - CPU
| ID |    10s |    60s |    15m | ID |    10s |    60s |    15m |
|  0 |   15.3 |   14.1 |   (na) |  1 |   20.8 |   18.0 |   (na) |
|  2 |   14.5 |   13.9 |   (na) |  3 |   23.3 |   21.1 |   (na) |
|  4 |   15.5 |   14.3 |   (na) |  5 |   13.0 |   11.8 |   (na) |
|  6 |   17.7 |   15.2 |   (na) |  7 |   17.7 |   15.8 |   (na) |
|  8 |   11.3 |   10.6 |   (na) |  9 |   12.8 |   11.6 |   (na) |
| 10 |   20.2 |   16.9 |   (na) | 11 |   26.8 |   23.9 |   (na) |
| 12 |   14.5 |   13.1 |   (na) | 13 |   17.3 |   16.4 |   (na) |
| 14 |   14.0 |   12.5 |   (na) | 15 |   11.9 |   10.9 |   (na) |
| 16 |   15.5 |   14.1 |   (na) | 17 |   13.7 |   12.7 |   (na) |
| 18 |   14.0 |   13.2 |   (na) | 19 |   22.9 |   20.9 |   (na) |
| 20 |   14.4 |   13.3 |   (na) | 21 |   20.1 |   18.1 |   (na) |
| 22 |   24.0 |   23.0 |   (na) |
```
It says I only have 23 threads (0-22), when I should have 24 (0-23). I have another instance like this, and the other one has 24 threads. I don't get why they are different. I did the exact same thing to both instances, when setting them up.

Help would be appreciated. Thanks!

**Comment by Spudz76** [comment]
What does `cat /proc/cpuinfo` and/or `hwloc-ls` say about cpus

**Comment by Spudz76** [comment]
cache size is wrong, must be different than the other instance?

46080 / 2048 = 22.5

needs 2048 * 24 = 49152 KB cache

**Comment by Yoshi24517** [comment]
On my 2nd instance, it has 56320 kb cache.

**Comment by Spudz76** [comment]
Yes, the one that won't use all cores needs at least 49152KB, and the one with "too much" cache will still only do max # of cores (so it is leaving some cache unused).

**Comment by Yoshi24517** [comment]
So... Is there any way to fix this?

**Comment by Spudz76** [comment]
Have Google Cloud set your cpus up correctly and identically / 24 cores with 2048KB cache each

**Comment by Yoshi24517** [comment]
I don't know where to check cache, but I will try to find it.

**Comment by Spudz76** [comment]
Yeah, nobody here probably ever seen what Google Cloud stuff looks like, much less used it, so all I can suggest is the setup is wrong and contact support or whoever.

**Comment by newmzy** [comment]
Can I ask an obvious question - is paying for a Google cloud upgraded account like this and cpu mining with it actually profitable, or is this just being done to mine some unprofitable coin in the hope of future mooning?
